packages/ngtools/webpack/src/compiler_host.ts

```diff
@@ -31,6 +34,7 @@ export class WebpackCompilerHost implements ts.CompilerHost {
   private _memoryHost: virtualFs.SyncDelegateHost;
   private _changedFiles = new Set<string>();
   private _basePath: Path;
+  private _ngccProcessor: NgccProcessor;
```

Comment:
Can you add some context here? 

Our previous approach was to make a standalone webpack plugin for `ngcc`. But we had to change that plan because the TS compilation itself (which happens before webpack bundling) could fail when looking up module typings, since the `.d.ts` files are also modified by `ngcc`. 

So now we hook into the TS module resolution and process them there.

Now that I think about this a bit more it begs the question: should `@angular/compiler-cli` do this directly? Is there some fundamental flaw with this approach we are missing, is it a matter of performance (since we don't do all entry points here), or is it just that we hadn't considered it before?

cc @petebacondarwin

Reply:
Are the d.ts files even relevant here though?  The only TS files should be application code and that shouldn’t be ivy specific.

Reply:
Dts files from external modules needs to be transformed for a ngtsc (Ivy) compilation.

Ngcc should only be run on external modules to transform dts and js files.

Reply:
Yes. but what TS code is being type checked that actually needs those d.ts files changed?

Reply:
Further, if the ivy compiler itself needs additional type information, can it not just do the analysis as needed itself?

Reply:
@petebacondarwin, can you chime in?

Reply:
Following the NGCC sync meeting, This current implementation is the preferred solution and NGTSC should not be aware of the existence of NGCC and these two compilation should be kept separate.